pages/lostfound.tsx

```diff
@@ -0,0 +1,180 @@
+import {React, useState, useEffect} from 'react';
+import { UserProvider } from '@auth0/nextjs-auth0';
+import Layout from '../lib/hoagie-ui/Layout';
+import Footer from '../components/Footer';
+import { InfoSignIcon, Text, Dialog, Heading, Alert, TextInputField, Pane, majorScale, minorScale, Spinner, EnvelopeIcon, ArrowLeftIcon, Button } from 'evergreen-ui'
+import Link from 'next/link';
+import Head from 'next/head';
+import { useUser } from '@auth0/nextjs-auth0'
+
+const senderNameDesc = `This will be the name of the sender displayed in the email. 
+It is recommended that you do not change this. However, you could change this to e.g.
+the name of the club advertising the event. Note that your real NetID will be included 
+at the bottom of the email regardless of your display name.`;
+
+export default function LostFound({ onSend, onError, errorMessage, success }) {
+  const { user, isLoading } = useUser();
+  if (isLoading)
+    return <Spinner />
+
+  const [name, setName] = useState('')
+  const [nameInvalid, setNameInvalid] = useState(false)
+  const [sender, setSender] = useState(user.name)
+  const [senderInvalid, setSenderInvalid] = useState(false)
+  const [filled, setFilled] = useState(false);
+  const [filledDesc, setFilledDesc] = useState(false);
+  const [desc, setDesc] = useState('')
+  const [descInvalid, setDescInvalid] = useState(false)
+  const [body, setBody] = useState('')
+  const [showConfirm, setShowConfirm] = useState(false)
+
+
+  useEffect( () => {
+    if (name == "") setFilled(true);
+    if (filled) setNameInvalid(name == "");
+    if (desc == "") setFilledDesc(true);
+    if (filledDesc) setDescInvalid(desc == "");
+    setSenderInvalid(sender == "");
+  }, [name, sender, desc]);
+
+  const MailForm = <Pane>
```

Comment:
Export most of this into a separate component called MailForm.tsx and use the component here instead.